istioctl/cmd/add-to-mesh.go

```diff
@@ -316,6 +316,10 @@ func injectSideCarIntoDeployment(client kubernetes.Interface, deps []appsv1.Depl
 	for _, dep := range deps {
 		log.Debugf("updating deployment %s.%s with Istio sidecar injected",
 			dep.Name, dep.Namespace)
+		if dep.Name == istioEgressgateway || dep.Name == istioIngressgateway {
```

Comment:
These names are not special, they just happen to be used in default install. Please avoid special logic based on these names. Prefer a check that looks for istio-proxy container?